index.js

```diff
@@ -2,14 +2,59 @@
 const execa = require('execa');
 const findVersions = require('find-versions');
 
-module.exports = (binary, options = {}) => {
-	return execa(binary, options.args || ['--version'])
-		.then(result => findVersions(result.stdout || result.stderr, {loose: true})[0])
-		.catch(error => {
+const knownBinaryArgs = new Map([
```

Comment:
```suggestion
const knownBinaryArguments = new Map([
```

Same with the others.